### PR TITLE
added error handling and parameter validation

### PR DESCRIPTION
**Pull request by mikethebeer**

### CODE REVIEW

File: php/index.php [hunk 1]
```diff
@@ -96,5 +118,15 @@
 
 });
 
+function error_response($app, $status, $message, $contenttype='application/json') {
+  $err = array("error" => $message, "status" => $status);
+  set_response($app, $status, json_encode($err), $contenttype);
+}
+
+function set_response($app, $status, $message, $contenttype='application/json') {
+  $app->response->headers->set('Con tent-Type', $contenttype);
```
Comment (chaudum) [inline php/index.php]: whitespace!

File: php/index.php [hunk 2]
```diff
@@ -56,29 +60,47 @@
...
-	$likeCount = $row['like_count'];
-	$likeCount++;
-	$qryU->bindParam(1, $likeCount);
-	$qryU->bindParam(2, $id);
-	$state = $qryU->execute();
-
-	header(CONTENT_TYPE_JSON);
-	echo json_encode(array('success' => $state));
+  if ($result) {
+    $qryU = $crate->prepare("UPDATE guestbook.posts SET like_count=? WHERE id=?");
```
Comment (chaudum) [inline php/index.php]:
```sql
UPDATE guestbook.posts SET like_count = like_count + 1 WHERE id=?
```

File: php/index.php [hunk 3]
```diff
@@ -56,29 +60,47 @@
...
+  	$qryU->bindParam(2, $id);
+  	$state = $qryU->execute();
+
+    if ($state) {
+      set_response($app, 204, json_encode(array('success' => $state)));
+    } else {
+      error_response($app, 500, 'update statement went wrong');
+    }
+  } else {
+    error_response($app, 500, 'post id={$id} not found');
```
Comment (chaudum) [inline php/index.php]: not found -> `404`
also error message differs from python implementation

File: php/index.php [hunk 4]
```diff
@@ -56,29 +60,47 @@
 });
 
 $app->delete('/posts/:id', function($id) use ($app, $crate) {
+  if ( empty($id) ) {
+    error_response($app, 400, 'post-id parameter is empty');
```
Comment (chaudum) [inline php/index.php]: check for correct error message (see python implementation)

File: php/index.php [hunk 5]
```diff
@@ -56,29 +60,47 @@
...
 	$qry = $crate->prepare("DELETE FROM guestbook.posts WHERE id=?");
 	$qry->bindParam(1, $id);
 	$state = $qry->execute();
 
-	header(CONTENT_TYPE_JSON);
-	echo json_encode(array('success' => $state));
+  if ($state) {
+    set_response($app, 204, json_encode(array('success' => $state)));
+  } else {
+    error_response($app, 500, 'post id={$id} not found');
```
Comment (chaudum) [inline php/index.php]: check for correct error message (see python implementation)

**Comment by joemoe** [comment]
agree with haudi, we should make sure the interface behaves just like the python app.

**Comment by chaudum** [comment]
please check response codes and error wording with python app and also test against the integration test suite!

**Pull request merged**